src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/versioncontrol/GitRepositoryFSManager.java

```diff
@@ -1,263 +1,330 @@
-		// clean up if possible after previous requests.
-		maintainNotUsedBranchPaths();
+    public GitRepositoryFSManager(String home, String project, List<String> repoUris,
+	    Map<String, String> defaultBranchNames) {
+	baseProjectUriPaths = new HashMap<String, String>();
+	baseProjectUriDefaultPaths = new HashMap<String, String>();
+	String baseProjectPath = home + File.separator + project;
+	for (int i = 0; i < repoUris.size(); i++) {
+	    baseProjectUriPaths.put(repoUris.get(i), baseProjectPath + File.separator + getShortHash(repoUris.get(i)));
+	    if (repoUris.size() == defaultBranchNames.keySet().size()) {
+		baseProjectUriDefaultPaths.put(repoUris.get(i), baseProjectUriPaths.get(repoUris.get(i))
+			+ File.separator + defaultBranchNames.get(repoUris.get(i)));
+	    } else {
+		baseProjectUriDefaultPaths.put(repoUris.get(i),
+			baseProjectUriPaths.get(repoUris.get(i)) + File.separator + "develop");
+	    }
 	}
+	// clean up if possible after previous requests.
+	maintainNotUsedBranchPaths();
+    }
 
-	/**
-	 * Returns target directory path for the default branch of the repository.
-	 *
-	 * @return absolute path to directory of the default branch
-	 */
-	public String getDefaultBranchPath() {
-		return baseProjectUriDefaultPath;
-	}
+    /**
+     * Returns target directory paths for the default branch of the repositories.
+     *
+     * @return absolute path to directories of the default branch
+     */
+    public Map<String, String> getDefaultBranchPaths() {
+	return baseProjectUriDefaultPaths;
+    }
 
-	/**
-	 * Makes branch's folder available in the temporary pool. It can significantly
-	 * contribute to improving the speed of check-outs of other branches as folder
-	 * renaming is not costly compared to copying.
-	 *
-	 * This mehod stays public as the developer might intend to release the folder
-	 * and not wait for maintenance strategy to trigger it.
-	 *
-	 * @param branchShortName
-	 *            branch name
-	 * @return null on failure, absolute path to new temporary directory
-	 */
-	public String releaseBranchDirectoryNameToTemp(String branchShortName) {
-		File oldDir = new File(getCheckoutPath(branchShortName));
+    /**
+     * Makes branch's folder available in the temporary pool. It can significantly
+     * contribute to improving the speed of check-outs of other branches as folder
+     * renaming is not costly compared to copying.
+     *
+     * This method stays public as the developer might intend to release the folder
+     * and not wait for maintenance strategy to trigger it.
+     *
+     * @param branchShortName branch name, Repository Uri for branch
+     * @return null on failure, absolute path to new temporary directory
+     */
+    public String releaseBranchDirectoryNameToTemp(String branchShortName, String repoUri) {
+	if (baseProjectUriPaths.containsKey(repoUri)) {
+	    String checkoutPath = getCheckoutPath(branchShortName, repoUri);
+	    if (!checkoutPath.equals("")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/UHD/cures-condec-jira/pullRequest?prid=4916975)